python_transport/wirepas_gateway/protocol/mqtt_wrapper.py

```diff
@@ -288,15 +299,33 @@ def publish(self, topic, payload, qos=1, retain=False) -> None:
         Args:
             topic: Topic to publish on
             payload: Payload
-            qos: Qos to use
-            retain: Is it a retain message
+            qos: Qos to use. Can be less than requested if broker does
```

Comment:
I think in general, we should just prevent usage the qos2 in all the code and have our own duplicate detection mechanism.
As far as I know, all broker support qos1